ludwig/hyperopt/execution.py

```diff
@@ -1178,37 +1179,42 @@ def run_experiment_trial(config, local_hyperopt_dict, checkpoint_dir=None):
                             sync_client.wait()
                         self._remove_partial_checkpoints(trial_path) # needed by get_best_checkpoint
                         best_model_path = analysis.get_best_checkpoint(trial_path.rstrip('/'))
-                        best_model = LudwigModel.load(
-                            os.path.join(best_model_path, 'model'),
-                            backend=backend,
-                            gpus=gpus,
-                            gpu_memory_limit=gpu_memory_limit,
-                            allow_parallel_threads=allow_parallel_threads,
-                        )
-                        if best_model.config[TRAINING]['eval_batch_size']:
-                            batch_size = best_model.config[TRAINING]['eval_batch_size']
-                        else:
-                            batch_size = best_model.config[TRAINING]['batch_size']
-                        try:
-                            eval_stats, _, _ = best_model.evaluate(
-                                dataset=validation_set,
-                                data_format=data_format,
-                                batch_size=batch_size,
-                                output_directory=trial_path,
-                                skip_save_unprocessed_output=skip_save_unprocessed_output,
-                                skip_save_predictions=skip_save_predictions,
-                                skip_save_eval_stats=skip_save_eval_stats,
-                                collect_predictions=False,
-                                collect_overall_stats=True,
-                                return_type='dict',
-                                debug=debug
+                        if best_model_path is not None:
```

Comment:
Not totally necessary for this PR, but the level of nesting here is a bit intimating. It might be worth abstracting out some of this logic into helper methods, or adding comments to describe stages of the overall flow.

Reply:
Good call out, thanks!  PR updated.